assets/global.js

```diff
@@ -553,6 +564,131 @@ class SliderComponent extends HTMLElement {
 
 customElements.define('slider-component', SliderComponent);
 
+class SlideshowComponent extends SliderComponent {
+  constructor() {
+    super();
+    this.sliderControlWrapper = this.querySelector('.slider-buttons');
+    this.bannerContents = this.slider.querySelectorAll('.banner__content');
+
+    if (!this.sliderControlWrapper) return;
+
+    this.sliderItemsToShow = Array.from(this.sliderItems).filter(element => element.clientWidth > 0);
+    this.sliderLastItem = this.sliderItemsToShow[this.sliderItemsToShow.length - 1];
+    if( this.sliderItemsToShow.length > 0) this.currentPage = 1;
+
+    this.sliderControlLinks = this.sliderControlWrapper.querySelectorAll('.slider-counter__link');
+    this.sliderControlLinksArray = Array.from(this.sliderControlLinks);
+    this.sliderAutoplay = this.slider.hasAttribute('data-autoplay');
+    this.mediaQueryMobile = window.matchMedia('(max-width: 749px)');
+    this.isPlaying = false;
+    this.autoPlayEnabled = false;
+
+    const resizeObserverControls = new ResizeObserver(entries => this.styleSlideshowControls());
+    resizeObserverControls.observe(this);
+    this.sliderControlLinks.forEach(link => link.addEventListener('click', this.linkToSlide.bind(this)));
+    this.slider.addEventListener('scroll', this.setSlideVisibility.bind(this));
+    this.setSlideVisibility();
+
+    if (!this.sliderAutoplay) return;
+    this.sliderAutoplayButton = this.sliderControlWrapper.querySelector('.slideshow__autoplay');
+    this.autoplaySpeed = this.slider.dataset.speed * 1000;
+
+    this.sliderAutoplayButton.addEventListener('click', this.autoPlayToggle.bind(this));
+    this.slider.addEventListener('mouseenter', this.autoplayFocusHandling.bind(this));
+    this.slider.addEventListener('mouseleave', this.autoplayFocusHandling.bind(this));
+    this.slider.addEventListener('focusin', this.autoplayFocusHandling.bind(this));
+    this.slider.addEventListener('focusout', this.autoplayFocusHandling.bind(this));
+    this.prevButton.addEventListener('focusout', this.autoplayFocusHandling.bind(this));
+    this.prevButton.addEventListener('focusin', this.autoplayFocusHandling.bind(this));
+    this.nextButton.addEventListener('focusin', this.autoplayFocusHandling.bind(this));
+    this.nextButton.addEventListener('focusout', this.autoplayFocusHandling.bind(this));
+    this.prevButton.addEventListener('mouseenter', this.autoplayFocusHandling.bind(this));
+    this.prevButton.addEventListener('mouseleave', this.autoplayFocusHandling.bind(this));
+    this.nextButton.addEventListener('mouseenter', this.autoplayFocusHandling.bind(this));
+    this.nextButton.addEventListener('mouseleave', this.autoplayFocusHandling.bind(this));
```

Comment:
This should probably be moved to a separate `setAutoplayEvents` function.

There's a lot happening on the `constructor` right now and I find that would help us know what is absolutely required and what isn't.